plugins/inputs/gnmi/gnmi.go

```diff
@@ -307,10 +311,15 @@ func New() telegraf.Input {
 	}
 }
 
-func init() {
+func (c *GNMI) init() error {
 	inputs.Add("gnmi", New)
 	// Backwards compatible alias:
 	inputs.Add("cisco_telemetry_gnmi", New)
+
+	if err := choice.CheckSlice(c.VendorSpecific, supportedExtensions); err != nil {
+		return fmt.Errorf("Unsupported vendor_specific option': %w", err)
+	}
+	return nil
```

Comment:
No, do not confuse this `init()` function with the `(c *GNMI) Init() error` method.. Those are 2 separate things.

Reply:
I changed to this : is it ok ?

func init() {
	inputs.Add("gnmi", New)
	// Backwards compatible alias:
	inputs.Add("cisco_telemetry_gnmi", New)
}

func (c *GNMI) Init() error {
	// Check vendor_specific options configured by user
	if err := choice.CheckSlice(c.VendorSpecific, supportedExtensions); err != nil {
		return fmt.Errorf("unsupported vendor_specific option: %w", err)
	}
	return nil
}

Reply:
Yes, I already approved it, you can close the conversation.